### PR TITLE
Switch DarkMode Styles according to localStorage.getItem('darkMode')

### LINKED ISSUE #400: Dark mode for map widget
Maps widget should switch to dark styles if in the dark mode. 

Reference: https://developers.google.com/maps/documentation/javascript/examples/style-array#maps_style_array-javascript

### PR DESCRIPTION
**Pull request by xrimpy**
Resource for Styling https://snazzymaps.com/

Fixes #400

### CODE REVIEW

File: frontend/src/Editor/Components/Map.jsx [hunk 1]
```diff
@@ -110,6 +192,7 @@ export const Map = function Map({
           mapContainerStyle={containerStyle}
           zoom={12}
           options={{
+            styles: localStorage.getItem('darkMode') === 'true' ? darkModeStyles : '',
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/Map.jsx:195]: Every widget will get `darkMode` as a parameter, we don't need to fetch from local storage. Can you please check `frontend/src/Editor/Components/Chart.jsx` and implement this logic in a similar way?

File: frontend/src/Editor/Components/Map/Map.jsx [hunk 2]
```diff
@@ -44,6 +45,8 @@ export const Map = function Map({
     height
   };
 
+  const darkModeStyles = require("./darkModeStyles.json");
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/Map/Map.jsx:48]: Can we move the imports to the top of the file?

Comment (xrimpy) [reply frontend/src/Editor/Components/Map/Map.jsx:48]: Extract it out as a JS array as in this example https://www.pluralsight.com/guides/load-and-render-json-data-into-react-components

File: frontend/src/Editor/Components/Map.jsx (hunk 1 above)
Comment (xrimpy) [reply frontend/src/Editor/Components/Map.jsx]: Using  Prop

**Pull request merged**